### PR TITLE
added example to Readme  for showing/hiding fields in view based on permitted_attributes

### PR DESCRIPTION
**Pull request by dmitrypol**

### CODE REVIEW

**Comment by dmitrypol** [comment]
happy to help.  I was wondering if anyone has a better solution?  I don't like putting logic in the view.  I was thinking of doing something with decorators to create CSS class to show/hide/disable the UI elements.

**Comment by damien** [comment]
@dmitrypol I think this is fine. I agree that in code you're working on you'd want better separation of concerns, but when you're talking about documentation/examples, simpler is better.

In your example, the important thing is to demonstrate that `policy(@post).permitted_attributes` can be queried to see if a subset of a record is authorized or not. Once people grok that they can query policies like that, I think it's reasonable to assume that they'll adjust their own implementation to match whatever format/style/separation is required by their own projects.

**Comment by deefour** [comment]
Thanks, this is an awesome use of `permitted_params` I never thought of.

**Comment by dmitrypol** [comment]
@jnicklas - I was wondering if you had a chance to consider this documentation PR?  Could be useful for other devs

File: README.md [hunk 1]
```diff
@@ -614,6 +614,20 @@ end
 
 If you have defined an action-specific method on your policy for the current action, the `permitted_attributes` helper will call it instead of calling `permitted_attributes` on your controller.
 
+To show/hide the UI element you can just call permitted_attributes method in the view:
+
+```ruby
+<%= simple_form_for(@post) do |f| %>
+...
+<% if policy(@post).permitted_attributes.include? :body %>
```
[the remark this answers is not in the document]
Comment (sekrett) [reply README.md]: I think it needs the post also, so: `permitted_attribute?(@post, :body)`.

**Comment by stephendolan** [comment]
Was working through this kind of problem in an app today, and came across a case that I believe will be quite common that this approach doesn't work for:

Let's say that there is a `Widget` model with a `creator_id` field. We do not want the user to submit that ID because we want to populate it in the `create` action of the controller with `current_user`.

However, we *do* want to show the `creator` to admins, and hide it from regular users.

In this case, `creator_id` would not be in `permitted_attributes`, so the check proposed here would return false. Has anyone come across a better way to implement something like this? My first thought is to literally implement a `visible_attributes` method to accompany `permitted_attributes`, but it would add a bit of bloat to my policies.

**Comment by Linuus** [comment]
Thanks, but I will close this PR for now. The readme is so long that it's hard to find information as it is. Perhaps we can add things like this to a Wiki or something in the future.

**Pull request closed without merging**